packages/scrapers-v2/src/classify/classify.ts

```diff
@@ -0,0 +1,47 @@
+import { ensureLengthAtLeast, loadHTML, parseText } from "../utils";
+import { CatalogEntryType, TypedCatalogEntry } from "./types";
+
+export const classifyCatalogEntries = async (
+  flattenedList: string[],
+): Promise<TypedCatalogEntry[]> => {
+  return await Promise.all(flattenedList.map(addTypeToUrl));
```

Comment:
I could be wrong by why await a promise when the return type is a promise. Is it because we are waiting for all async functions to finish within the `Promise.all()`?

Reply:
if the promise ends up rejecting, then the stacktrace will originate from this function, because the promise was `await`ed here. if we don't include await, and the promise rejects, the stacktrace will originate from wherever the promise happens to be awaited, making the code harder to debug

Reply:
not awaiting my promises before returning always gets me :(

Reply:
tbh im still in the guess and check state even after swe ;))

Reply:
lmfaoo